### PR TITLE
PMM-6038 PT summary.

### PR DESCRIPTION
**Pull request by JiriCtvrtka**

### CODE REVIEW

File: services/management/grpc/actions_server.go [hunk 1]
```diff
@@ -266,3 +267,31 @@ func (s *actionsServer) CancelAction(ctx context.Context, req *managementpb.Canc
 
 	return &managementpb.CancelActionResponse{}, nil
 }
+
+// StartPTSummaryAction starts pt-summary action.
+//nolint:lll,dupl
+func (s *actionsServer) StartPTSummaryAction(ctx context.Context, req *managementpb.StartPTSummaryActionRequest) (*managementpb.StartPTSummaryActionResponse, error) {
+	agent, err := models.FindAgentByID(s.db.Querier, req.PmmAgentId)
```
[the remark this answers is not in the document]
Comment (BupycHuk) [reply services/management/grpc/actions_server.go:274]: No, we should check for type.

File: services/management/grpc/actions_server.go [hunk 2]
```diff
@@ -266,3 +267,33 @@ func (s *actionsServer) CancelAction(ctx context.Context, req *managementpb.Canc
...
+	if err != nil {
+		return nil, err
+	}
+	if agent.AgentID != req.PmmAgentId {
```
Comment (BupycHuk) [inline services/management/grpc/actions_server.go:278]: this always will be true

**Review by BupycHuk — CHANGES_REQUESTED** [review]

File: services/management/grpc/actions_server.go [hunk 3]
```diff
@@ -266,3 +267,33 @@ func (s *actionsServer) CancelAction(ctx context.Context, req *managementpb.Canc
...
+
+// StartPTSummaryAction starts pt-summary action.
+//nolint:lll,dupl
+func (s *actionsServer) StartPTSummaryAction(ctx context.Context, req *managementpb.StartPTSummaryActionRequest) (*managementpb.StartPTSummaryActionResponse, error) {
+	agent, err := models.FindAgentByID(s.db.Querier, req.PmmAgentId)
+	if err != nil {
+		return nil, err
+	}
+	if agent.AgentType != models.PMMAgentType {
+		return nil, fmt.Errorf("pmm-agent has type %s not %s", agent.AgentType, models.PMMAgentType)
```
Comment (BupycHuk) [inline services/management/grpc/actions_server.go:279]: Couldn't find pmm-agent to run action

**Review by BupycHuk — APPROVED** [review]
LGTM, just one comment.
Please request review from @percona-csalguero for all PRs.

File: services/management/grpc/actions_server.go [hunk 4]
```diff
@@ -266,3 +269,33 @@ func (s *actionsServer) CancelAction(ctx context.Context, req *managementpb.Canc
 
 	return &managementpb.CancelActionResponse{}, nil
 }
+
+// StartPTSummaryAction starts pt-summary action.
+//nolint:lll
+func (s *actionsServer) StartPTSummaryAction(ctx context.Context, req *managementpb.StartPTSummaryActionRequest) (*managementpb.StartPTSummaryActionResponse, error) {
+	agent, err := models.FindAgentByID(s.db.Querier, req.PmmAgentId)
```
Comment (BupycHuk) [inline services/management/grpc/actions_server.go:276]: find pmm-agents by field RunsOnNodeID which should be equal to req.NodeID. For it use `models.FindPMMAgentsRunningOnNode`.
Then check that passed pmm-agent id in the list by using `models.FindPmmAgentIDToRunAction`.

File: services/management/grpc/actions_server.go [hunk 5]
```diff
@@ -266,3 +269,33 @@ func (s *actionsServer) CancelAction(ctx context.Context, req *managementpb.Canc
...
+	if err != nil {
+		return nil, err
+	}
+	if agent.AgentType != models.NodeExporterType {
```
Comment (BupycHuk) [inline services/management/grpc/actions_server.go:280]: This check can be removed.

File: services/management/grpc/actions_server.go [hunk 6]
```diff
@@ -266,3 +269,33 @@ func (s *actionsServer) CancelAction(ctx context.Context, req *managementpb.Canc
...
+		return nil, fmt.Errorf("couldn't find pmm-agent to run action")
+	}
+	if pointer.GetString(agent.NodeID) != req.NodeId {
```
Comment (BupycHuk) [inline services/management/grpc/actions_server.go:283]: This check can be removed.

File: services/management/grpc/actions_server.go [hunk 7]
```diff
@@ -266,3 +269,33 @@ func (s *actionsServer) CancelAction(ctx context.Context, req *managementpb.Canc
 
 	return &managementpb.CancelActionResponse{}, nil
 }
+
+// StartPTSummaryAction starts pt-summary action.
+//nolint:lll
+func (s *actionsServer) StartPTSummaryAction(ctx context.Context, req *managementpb.StartPTSummaryActionRequest) (*managementpb.StartPTSummaryActionResponse, error) {
```
Comment (BupycHuk) [inline services/management/grpc/actions_server.go:275]: Place this method above CancelAction

**Review by BupycHuk — CHANGES_REQUESTED** [review]

File: Gopkg.toml [hunk 8]
```diff
@@ -16,7 +16,7 @@ required = [
 
 [[constraint]]
   name = "github.com/percona/pmm"
-  branch = "PMM-2.0"
+  branch = "PMM-6038-pt-summary"
```
Comment (BupycHuk) [inline Gopkg.toml:19]: TODO:

**Review by BupycHuk — APPROVED** [review]

**Comment by JiriCtvrtka** [comment]
FB: https://github.com/Percona-Lab/pmm-submodules/pull/1065

**Pull request merged**